### PR TITLE
Check and notify for updates

### PR DESCRIPTION
**Pull request by vadi2**
#### Brief overview of PR changes/additions
Check and notify for updates.
#### Motivation for adding to Hammer
So people don't get stuck on old versions with outdated validators.
#### Other info (issues closed, discussion etc)
Doesn't automatically update Hammer itself - just shows a link to the user to go update. It was too much work to do automatic updates for a .NET Core application on all 3 platforms and why the project stalled :( so this is better than nothing and we can finally have the first release!

### CODE REVIEW

**Comment by vadi2** [comment]
Ooh... thanks bot.

File: Program.cs [hunk 1]
```diff
@@ -852,6 +865,84 @@ public void CancelValidation()
...
+            if (latestVersion > currentVersion)
+            {
+                Console.WriteLine($"Newer version available: {latestVersion}");
+                this.ActivateSignal("updateAvailable", latestVersion.ToString());
+            } else {
+                Console.WriteLine($"No newer version available; latest released on Github is {latestVersion}.");
+            }
+        }
+
+        public Version GetLatestVersion(string repoTagsRaw)
```
Comment (pieter-edelman-nictiz) [inline Program.cs]: Maybe 'ExtractLatestVersion' is a better name?

File: Program.cs [hunk 2]
```diff
@@ -852,6 +865,84 @@ public void CancelValidation()
...
+                return;
+            }
+
+            var latestVersion = GetLatestVersion(tags);
+
+            if (latestVersion == null) {
+                return;
+            }
+
+            if (latestVersion > currentVersion)
```
Comment (pieter-edelman-nictiz) [inline Program.cs]: Does this comparison work? From what I can see in https://docs.microsoft.com/en-us/dotnet/api/system.version?view=netframework-4.8#comparing-version-objects, you need to use the CompareTo method.

File: Program.cs [hunk 3]
```diff
@@ -407,6 +414,12 @@ private static MatchCollection ExtractLocation(string rawLocation)
             return Regex.Matches(rawLocation, pattern);
         }
 
+        private static MatchCollection ExtractReleaseVersion(string rawVersion)
```
Comment (pieter-edelman-nictiz) [inline Program.cs]: Perhaps this separate function is a bit overhead (just two lines, called just once)

**Comment by vadi2** [comment]
Ah man! I did not notice that you left feedback. I'll send in a follow-up PR.

File: Program.cs (hunk 2 above)
Comment (vadi2) [reply Program.cs]: Yes it seemed to in my testing... I got it from an SO answer, but I tested the different values. I'll ask a .NET expert.

File: Program.cs (hunk 3 above)
Comment (vadi2) [reply Program.cs]: We do this so the regex isn't compiled all the time, remember

**Pull request merged**